core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -149,6 +233,62 @@ public SqlToConnectorExpressionTranslator(Map<NodeRef<Expression>, Type> types)
             return Optional.of(new Constant(null, typeOf(node)));
         }
 
+        @Override
+        protected Optional<ConnectorExpression> visitFunctionCall(FunctionCall node, Void context)
+        {
+            if (!node.getFilter().isEmpty() || !node.getOrderBy().isEmpty() || !node.getWindow().isEmpty() || node.isDistinct()) {
+                return Optional.empty();
+            }
+
+            String functionName = ResolvedFunction.extractFunctionName(node.getName());
+
+            List<ConnectorExpression> arguments = new ArrayList<>();
+            for (Expression argumentExpression : node.getArguments()) {
+                if (argumentExpression instanceof SymbolReference || argumentExpression instanceof FunctionCall || argumentExpression instanceof StringLiteral) {
+                    Optional<ConnectorExpression> argument = process(argumentExpression);
+                    if (argument.isEmpty()) {
+                        return Optional.empty();
+                    }
+                    arguments.add(argument.get());
+                }
+            }
+
+            // Not sure if it can be relevant (maybe always Ignore \ Respect in applyFilter functions?)
+            Optional<Function.NullTreatment> nullTreatment;
+            if (node.getNullTreatment().isEmpty()) {
+                nullTreatment = Optional.empty();
+            }
+            else if (node.getNullTreatment().get() == FunctionCall.NullTreatment.IGNORE) {
+                nullTreatment = Optional.of(Function.NullTreatment.IGNORE);
+            }
+            else if (node.getNullTreatment().get() == FunctionCall.NullTreatment.RESPECT) {
+                nullTreatment = Optional.of(Function.NullTreatment.RESPECT);
+            }
+            else {
+                throw new RuntimeException("Unknown NullTreatment value " + node.getNullTreatment().get().name());
+            }
+
+            // TODO: replace after resolving the issue with FunctionPushdown
+            // return Optional.of(new Function(typeOf(node), functionName, arguments, nullTreatment));
+            return Optional.empty();
+        }
+
+        @Override
+        protected Optional<ConnectorExpression> visitLikePredicate(LikePredicate node, Void context)
```

Comment:
This should never occur. Like expressions are desugared into function calls of an internal function.

Reply:
It occurs for example when running `io.trino.plugin.memory.TestMemorySmoke#testLikeConnectorExpressionPushdown`. That's because `DesugarLike` is executed after `PushPredicateIntoTableScan` at `PlanOptimizers`.